test/scripts/gcp-bench-cluster.sh

```diff
@@ -5,47 +5,46 @@ set -o nounset
 set -o pipefail
 
 readonly OPERATION=$1
-readonly CLUSTER_NAME=${CLUSTER_NAME:-contour-bench-$$}
-readonly CLUSTER_ZONE=${CLUSTER_ZONE:-us-east1-d}
-readonly CLUSTER_VERSION=${CLUSTER_VERSION:-latest}
-readonly NODE_MACHINE_TYPE=${NODE_MACHINE_TYPE:-e2-standard-4}
-readonly NUM_APP_POOL_NODES=${NUM_APP_POOL_NODES:-10}
+readonly CONTOUR_BENCH_CLUSTER_NAME=${CONTOUR_BENCH_CLUSTER_NAME:-contour-bench-$$}
+readonly CONTOUR_BENCH_CLUSTER_ZONE=${CONTOUR_BENCH_CLUSTER_ZONE:-us-east1-d}
+readonly CONTOUR_BENCH_CLUSTER_VERSION=${CONTOUR_BENCH_CLUSTER_VERSION:-latest}
+readonly CONTOUR_BENCH_NODE_MACHINE_TYPE=${CONTOUR_BENCH_NODE_MACHINE_TYPE:-e2-standard-4}
+readonly CONTOUR_BENCH_NUM_APP_POOL_NODES=${CONTOUR_BENCH_NUM_APP_POOL_NODES:-10}
 
 # TODO: If credentials available in environment, log in.
 
 function deploy() {
-  gcloud container clusters create ${CLUSTER_NAME} \
-    --cluster-version=${CLUSTER_VERSION} \
-    --zone=${CLUSTER_ZONE} \
+  num_nodes=$CONTOUR_BENCH_NUM_APP_POOL_NODES
+  ((num_nodes+=1))
```

Comment:
what's this for?

Reply:
previously i had two node pools to separate the worker nodes from contour nodes but when you add a node pool GKE always automatically repairs your cluster so its not usable really right after its created

this way you ask for some number of worker nodes and we add 1 which contour is deployed to (to separate contour and possibly allow us to add more specific resource constraints to test it in isolation, rather than it run on a node with other variable workloads)

Reply:
Ah, makes sense, a short comment here would be helpful.

Reply:
:+1: